src/Crypto/Lol/Applications/SymmSHE.hs

```diff
@@ -276,7 +276,7 @@ switch hint c = rescaleLinearMSD $ untag $ knapsack <$>
 
 -- | Constraint synonym for key switching.
 type KeySwitchCtx gad t m' zp zq zq' =
-  (ToSDCtx t m' zp zq,
+  (ToSDCtx t m' zp zq, NFData (Cyc t m' zq'),
```

Comment:
Then something is wrong at a higher level.  For example, if `CElt` includes an `NFData` constraint on both the element type and its `CRTExt`, then I don't think we need this constraint.  I think `CElt` should have all constraints that could ever be needed for the interface to `Cyc`.